src/ImageSharp/Processing/Processors/Convolution/BokehBlurProcessor.cs

```diff
@@ -0,0 +1,534 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.ParallelUtils;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Primitives;
+using SixLabors.Memory;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Convolution
+{
+    /// <summary>
+    /// Applies bokeh blur processing to the image.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal class BokehBlurProcessor<TPixel> : ImageProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        /// <summary>
+        /// The maximum size of the kernel in either direction.
+        /// </summary>
+        private readonly int kernelSize;
+
+        /// <summary>
+        /// The number of components to use when applying the bokeh blur
+        /// </summary>
+        private readonly int componentsCount;
+
+        /// <summary>
+        /// The kernel parameters to use for the current instance (a: X, b: Y, A: Z, B: W)
+        /// </summary>
+        private readonly Vector4[] kernelParameters;
+
+        /// <summary>
+        /// The kernel components for the current instance
+        /// </summary>
+        private readonly Complex64[][] kernels;
+
+        /// <summary>
+        /// The scaling factor for kernel values
+        /// </summary>
+        private readonly float kernelsScale;
+
+        /// <summary>
+        /// The mapping of initialized complex kernels and parameters, to speed up the initialization of new <see cref="BokehBlurProcessor{TPixel}"/> instances
+        /// </summary>
+        private static readonly Dictionary<(int, int), (Vector4[], float, Complex64[][])> Cache = new Dictionary<(int, int), (Vector4[], float, Complex64[][])>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Convolution.BokehBlurProcessor{TPixel}"/> class.
+        /// </summary>
+        /// <param name="radius">
+        /// The 'radius' value representing the size of the area to sample.
+        /// </param>
+        /// <param name="components">
+        /// The number of components to use to approximate the original 2D bokeh blur convolution kernel.
+        /// </param>
+        /// <param name="gamma">
+        /// The gamma highlight factor to use to further process the image.
+        /// </param>
+        public BokehBlurProcessor(int radius = 32, int components = 2, float gamma = 3)
```

Comment:
I've used default parameters there because especially for the `components` and `gamma` parameters, I wouldn't expect most users to be able to pick the right values on their own, as they require at least some knowledge of how the effect works to be undestoood at all.
`2` components and a gamma of `3` provide a good compromise for quality and speed of execution, that should work fine in most cases.
What if I only made those two optional, and left the `radius` parameter with no default value?

Reply:
I changed this during the rebase